### ISSUE TITLE
bug: use kubernetes dicovery start failed

### ISSUE THREAD

**Issue by myoss** [apache/apisix#7473]
### Current Behavior

apisix use kubernetes dicovery start failed

### Error Logs

 kubectl -n ingress-apisix logs -f apisix-bd9687cc7-68vvr

```
/usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua init
/usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua init_etcd
2022/07/15 17:25:24 [warn] 1#1: low address bits of 127.0.0.1/24 are meaningless in /usr/local/apisix/conf/nginx.conf:287
nginx: [warn] low address bits of 127.0.0.1/24 are meaningless in /usr/local/apisix/conf/nginx.conf:287
nginx: [error] init_by_lua error: /usr/local/openresty/lualib/ngx/process.lua:5: unsupported subsystem: stream
stack traceback:
        [C]: in function 'error'
        /usr/local/openresty/lualib/resty/core/base.lua:223: in function 'allows_subsystem'
        /usr/local/openresty/lualib/ngx/process.lua:5: in main chunk
        [C]: in function 'require'
        /usr/local/apisix/apisix/discovery/kubernetes/init.lua:27: in main chunk
        [C]: in function 'require'
        /usr/local/apisix/apisix/discovery/init.lua:28: in main chunk
        [C]: in function 'require'
        /usr/local/apisix/apisix/upstream.lua:19: in main chunk
        [C]: in function 'require'
        /usr/local/apisix/apisix/http/service.lua:18: in main chunk
        [C]: in function 'require'
        /usr/local/apisix/apisix/init.lua:34: in main chunk
        [C]: in function 'require'
        init_by_lua:3: in main chunk
```

### Steps to Reproduce

vi charts/apisix/values.yaml

```
discovery:
  enabled: true
  registry:
    kubernetes:
      service:
        # apiserver schema, options [http, https]
        schema: https #default https

        # apiserver host, options [ipv4, ipv6, domain, environment variable]
        host: ${KUBERNETES_SERVICE_HOST} #default ${KUBERNETES_SERVICE_HOST}

        # apiserver port, options [port number, environment variable]
        port: ${KUBERNETES_SERVICE_PORT}  #default ${KUBERNETES_SERVICE_PORT}

      client:
        # serviceaccount token or token_file
        token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
      namespace_selector:
        # only save endpoints with namespace equal default
        equal: app
```

helm upgrade apisix charts/apisix  --namespace ingress-apisix

 kubectl -n ingress-apisix logs -f apisix-bd9687cc7-68vvr

### Environment

- APISIX version (run `apisix version`): use latest code [https://github.com/apache/apisix/commit/a04964e2a3ad1430152818ed3dbb8130438de877](https://github.com/apache/apisix/commit/a04964e2a3ad1430152818ed3dbb8130438de877), build docker images
- Operating system (run `uname -a`): `Linux iZbp18of4fhb66ti309tdmZ 3.10.0-1160.31.1.el7.x86_64 #1 SMP Thu Jun 10 13:32:12 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by myoss** [comment]
```
bash-5.1# openresty -v
nginx version: openresty/1.19.3.2
```

**Comment by tokers** [comment]
@myoss How did you install apisix? You need to use the suggest way (from the apisix website, see  https://apisix.apache.org/docs/apisix/how-to-build/#installation-via-rpm-offline-package for details) build apisix-base.

**Comment by myoss** [comment]
use `HELM`, build docker image [https://github.com/apache/apisix-docker/blob/release/apisix-2.13.0/alpine-local/Dockerfile](https://github.com/apache/apisix-docker/blob/release/apisix-2.13.0/alpine-local/Dockerfile)

**Comment by tokers** [comment]
Looks like the `ngx.process` module was loaded in the stream system and it's not supported.

**Comment by tzssangglass** [comment]
maybe we should check here: https://github.com/apache/apisix/blob/eaaad397bfb4899f7f39d585309275c54103620d/apisix/discovery/kubernetes/init.lua#L27

change here to: 

```lua
local is_http           = ngx.config.subsystem == "http"

if is_http then
    process = require("ngx.process")
end
```

But I'm still curious as to what our test cases should cover for this scenario, see: https://github.com/apache/apisix/blob/eaaad397bfb4899f7f39d585309275c54103620d/t/cli/test_kubernetes.sh#L20-L48

maybe we should check `make run` here?

**Comment by tokers** [comment]
> maybe we should check here:
> 
> https://github.com/apache/apisix/blob/eaaad397bfb4899f7f39d585309275c54103620d/apisix/discovery/kubernetes/init.lua#L27
> 
> change here to:
> 
> ```lua
> local is_http           = ngx.config.subsystem == "http"
> 
> if is_http then
>     process = require("ngx.process")
> end
> ```
> 
> But I'm still curious as to what our test cases should cover for this scenario, see:
> 
> https://github.com/apache/apisix/blob/eaaad397bfb4899f7f39d585309275c54103620d/t/cli/test_kubernetes.sh#L20-L48
> 
> we should check `make run` here?

I think so, if the `make run` doesn't carry any other side effects.

**Comment by tzssangglass** [comment]
@tao12345666333 Can you do a quick verification for this?

**Comment by tao12345666333** [comment]
this project https://github.com/apache/apisix-ingress-controller

**Comment by zhixiongdu027** [comment]
@myoss 

looks like you are trying to use Kubernetes Discovery in the Stream Module. Right ?

The reason for your problem is that "ngx.process"  does not support Stream Module in openresty/1.19.3.2

This problem can only be solved by upgrading the Openresty version( >=1.19.9.1 ) included with APISIX, 
or by using other methods instead of Kubernetes Discovery,  For example, apisix-ingress-controller mentioned by @tao12345666333
